scripts/parseDeployments.js

```diff
@@ -0,0 +1,136 @@
+  console.log('-------------------------------------------+------------------------------');
+  let tokenSum = toBN(0);
+  for (const address of addressList) {
+    const balance = await wfairtoken.balanceOf(address);
+    console.log(address + ' | ' + balance.toString().padStart(28));
+    tokenSum = tokenSum.add(balance);
+  }
+  console.log('Tokens accounted for: ' + tokenSum.toString());
+  if (tokenSum.eq(totalSupply)) {
+    console.log('All tokens accounted for.');
+  } else {
+    console.error('❌ ERROR: tokens unaccounted for: ' + (totalSupply.sub(tokenSum).toString()));
+  }
+
+  // Checking initial deployment of lock
+  console.log('\nChecking token locks');
+  for (const lock of actions.locks) {
+    console.log('\nToken lock contract name: ' + lock.name);
+    console.log('Token lock contract address: ' + lock.address);
+    // TODO: call state of contract view function
+    console.log('State of contract: ');
+    // connect to lock contract
+    const LockContract = await hre.ethers.getContractFactory('TokenLock');
+    const lockContract = LockContract.attach(lock.address);
+    // loop through each address we have on record, checking total, unlocked and vested quantities
+    const retrievedContractData = [];
+    console.log('Account                                    |' +
+      '                        Total |' +
+      '                       Vested |' +
+      '                     Unlocked');
+    console.log('-------------------------------------------+' +
+      '------------------------------+' +
+      '------------------------------+' +
+      '------------------------------');
+    for (const stakeholder of lock.parameters[5]) {
+      const totalTokens = await lockContract.totalTokensOf(stakeholder);
+      const vestedTokens = await lockContract.tokensVested(stakeholder, blockchainTime());
+      const unlockedTokens = await lockContract.unlockedTokensOf(stakeholder);
+      console.log(stakeholder + ' | ' +
+        totalTokens.toString().padStart(28) + ' | ' +
+        vestedTokens.toString().padStart(28) + ' | ' +
+        unlockedTokens.toString().padStart(28));
+      // store data for final tally comparisons
+      retrievedContractData.push([stakeholder, totalTokens, vestedTokens, unlockedTokens]);
+    }
+    // get current number of WFAIR tokens in lock
+    const lockBalance = await wfairtoken.balanceOf(lock.address);
+    console.log('Current WFAIR balance of contract: ' + lockBalance.toString());
+    // check that lock balance = sum of totals minus unlocked
+    let totalCalc = toBN(0);
+    for (const entry of retrievedContractData) {
+      totalCalc = totalCalc.add(entry[1]).sub(entry[3]);
+    }
+    console.log('Required funds for token lock    : ' + totalCalc.toString());
+    if (totalCalc.eq(lockBalance)) {
+      console.log('The token lock is fully funded and able to cover payouts');
+    } else {
+      console.log('❌ ERROR: token amount in contract does not match required funding');
+    }
```

Comment:
check if contract is funded as well

Reply:
Ah, you mean "check if contract is Funded as well" ...